adapters/p2p2consensus/proposal.go

```diff
@@ -0,0 +1,83 @@
+package p2p2consensus
+
+import (
+	"math/big"
+
+	"github.com/Masterminds/semver/v3"
+	consensus "github.com/NethermindEth/juno/consensus/types"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	p2pconsensus "github.com/NethermindEth/juno/p2p/proto/consensus/consensus"
+	"github.com/NethermindEth/juno/utils"
+	common "github.com/starknet-io/starknet-p2pspecs/p2p/proto/common"
+)
+
+func U128ToFelt(u *common.Uint128) *felt.Felt {
+	lowBig := new(big.Int).SetUint64(u.Low)
+	highBig := new(big.Int).SetUint64(u.High)
+	highBig.Lsh(highBig, 64) //nolint:mnd
+	return new(felt.Felt).SetBigInt(highBig.Or(highBig, lowBig))
+}
+
+func AdaptProposalInit(msg *p2pconsensus.ProposalInit) consensus.ProposalInit {
+	return consensus.ProposalInit{
+		BlockNum: msg.BlockNumber,
+		Proposer: *new(felt.Felt).SetBytes(msg.Proposer.Elements),
+	}
+}
+
+func AdaptBlockInfo(msg *p2pconsensus.BlockInfo) *consensus.BlockInfo {
+	return &consensus.BlockInfo{
+		BlockNumber:       msg.BlockNumber,
+		Builder:           *new(felt.Felt).SetBytes(msg.Builder.Elements),
+		Timestamp:         msg.Timestamp,
+		L2GasPriceFRI:     *U128ToFelt(msg.L2GasPriceFri),
+		L1GasPriceWEI:     *U128ToFelt(msg.L1DataGasPriceWei),
+		L1DataGasPriceWEI: *U128ToFelt(msg.L1DataGasPriceWei),
+		EthToStrkRate:     *U128ToFelt(msg.EthToStrkRate),
+		L1DAMode:          core.L1DAMode(msg.L1DaMode),
+	}
+}
+
+func AdaptProposalCommitment(msg *p2pconsensus.ProposalCommitment) consensus.ProposalCommitment {
+	return consensus.ProposalCommitment{
+		BlockNumber: msg.BlockNumber,
+		Builder:     *new(felt.Felt).SetBytes(msg.Builder.Elements),
+
```

Comment:
nit: feel free to ignore, but I think removing blank lines between fields aligns the values, which help readability a bit.